src/InteractiveWindow/Editor/InteractiveWindow_UIThread.cs

```diff
@@ -1840,19 +1981,38 @@ public bool TrySubmitStandardInput()
             private void SubmitStandardInput()
             {
                 AppendLineNoPromptInjection(_window._standardInputBuffer);
-                _window._inputValue = new SnapshotSpan(_window._standardInputBuffer.CurrentSnapshot, Span.FromBounds(_window._stdInputStart.Value, _window._standardInputBuffer.CurrentSnapshot.Length));
-                _window._inputEvent.Set();
+                var inputSpan = new SnapshotSpan(_window._standardInputBuffer.CurrentSnapshot, GetStandardInputSpan());
+                History.Add(inputSpan);
+                SetStandardInputValue(inputSpan);
+
+                MakeStandardInputReadonly();
+
+                // Subsequent input should appear after the input span we just finished.
+                NewOutputBuffer();
+
+                if (State == State.WaitingForInputAndReadingStandardInput)
+                {
+                    PrepareForInput(); // Will update State.
+                }
+                else
+                {
+                    State = GetStateBeforeReadingStandardInput(State);
+                }
             }
 
             private bool InStandardInputRegion(SnapshotPoint point)
             {
-                if (_window._stdInputStart == null)
+                if (!ReadingStandardInput)
                 {
                     return false;
                 }
 
-                var stdInputPoint = _window.GetPositionInStdInputBuffer(point);
-                return stdInputPoint != null && stdInputPoint.Value.Position >= _window._stdInputStart.Value;
+                var standardInputPoint = _window.GetPositionInStandardInputBuffer(point);
+                if (!standardInputPoint.HasValue) return false;
```

Comment:
@KevinH-MS I must have been thinking of other code.  This file clearly favors multi-line ifs.